### PR TITLE
Improve development instructions 

### PR DESCRIPTION
**Pull request by raquelalegre**
This contains a more updated and detailed version of the development instructions in the production repository.

### CODE REVIEW

**Comment by raquelalegre** [comment]
These are aimed mainly at @clairbarnes and @JamieJQuinn so small fixes and text changes can continue to be done and deployed after I have left the project. Take your time to make sure you understand these and ask for any clarification.

**Review by giordano — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,18 +1,102 @@
...
-To submit new feature requests or issue, please do so [in the production repository](https://github.com/UCL/UKCORDEX-plot-explorer/issues/new).
+To submit new feature requests or issues, please do so [in the production repository](https://github.com/UCL/UKCORDEX-plot-explorer/issues/new). Issues are disabled on this one.
 
-## Deploy website locally
+## Development guide
+
+### Dependencies
+This website is built using React. You will need:
+
+* npm (Last tested with npm 8.0.0)
```
Comment (JamieJQuinn) [inline README.md]: Is it more useful to list which version the CI uses during its test deployment? It looks like we're running node version 16 but I don't know how the versions of npm and node relate...

File: README.md [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
+
+This project has the following dependencies:
+* create-react-app
```
Comment (JamieJQuinn) [inline README.md]: The list of the dependencies in the README could diverge from the [actual dependencies](https://github.com/UCL/UKCORDEX-plot-explorer-dev/blob/2184e1be0f3a43f7028918169e57fa4aa563399f/package.json#L6) pretty quickly, do we want to just link to `package.json`?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,102 @@
...
+### Continuous Integration and Deployment
+
+There are [3 GH Actions workflows](https://github.com/UCL/UKCORDEX-plot-explorer-dev/tree/development/.github/workflows) configured:
+* Build and test: Builds the app and runs test for 3 different versions of node
+* Deploy to development: Builds the app and deploys to the development URL (https://github-pages.ucl.ac.uk/UKCORDEX-plot-explorer-dev)
+* Deploy to production: Builds the app and deploys to the production URL (https://github-pages.ucl.ac.uk/UKCORDEX-plot-explorer)
+
+For CD we are using [JamesIves/github-pages-deploy-action](https://github.com/JamesIves/github-pages-deploy-action).
+
+We are using [JamesIves/github-pages-deploy-action](https://github.com/JamesIves/github-pages-deploy-action) to deploy automatically using GH Actions, and [the npm gh-pages module](https://github.com/tschaub/gh-pages) to deploy
```
Comment (JamieJQuinn) [inline README.md]: Is this a duplicate of line 32?

File: README.md [hunk 4]
```diff
@@ -1,18 +1,102 @@
...
+
+### Workflow guidelines
+To work on a new feature or bug fix, please do as follows:
+1. Create a new branch, e.g. `feature/about_us_section`.
```
Comment (JamieJQuinn) [inline README.md]: Without an explicit contributor's guide, should we explicitly say we want branches named `<feature/bug>/<description>`?

File: README.md [hunk 5]
```diff
@@ -1,18 +1,102 @@
...
 ```
 npm install [--force]
 npm start
 ```
 
-Note `--force` might be needed when dependencies prevent the project from building, but it's preferable to investigate the dependency issues and resolve them avoiding force instal.
+You should see now a message in your console indicating your web is being
+deployed in localhost:3000 (or different port if 3000 is busy). Note that you don't need to kill this process and deploy it again if you want to work on changes to the code - they will update in the locally deployed app immediately after saving.
+
+Also note `--force` might be needed when dependencies prevent the project from building, but it's preferable to investigate the dependency issues with `npm audit` and resolve them avoiding force instal.
```
Comment (JamieJQuinn) [inline README.md]:
```suggestion
Also note `--force` might be needed when dependencies prevent the project from building, but it's preferable to investigate the dependency issues with `npm audit` and resolve them avoiding force install.
```

**Review by JamieJQuinn — CHANGES_REQUESTED** [review]
Beautiful! Suggested some tiny changes. Feel free to ignore if too nit-picky.

File: README.md (hunk 1 above)
Comment (raquelalegre) [reply README.md]: There isn't a straight forward answer to this :)

The version of `npm` used in the CI is whichever the `ubuntu-latest` comes with. And I guess that'll change in the future, unless I force one version in particular, which might not be a good idea. The local npm version I've used was 8.0.0 at the time I wrote that message, but I now have 8.10.0 😬 Perhaps I should just remove the parenthesis.

The more important version is Node.js's. We are running one workflow for the tests which tries on several versions, and then for the deployment workflow we are using the latest, which is 16 as you say. We can't deploy on a matrix as they would all be deploying to the same GitHub Pages URL, and we wouldn't know which one of the three we'd be looking at! We could do a work around and deploy to three different paths within the same URL, but I didn't have enough time for that :(

The correlation between NPM and Node.js is also not straight forward. They are completely separate entities, but there are a number of versions of Node.js that a given version of NPM wouldn't support. 

I'll update the text in the parenthesis, but feel free to let me know if it's not clear and I should just remove it.

File: README.md (hunk 2 above)
Comment (raquelalegre) [reply README.md]: Yep, that's a good idea, I'll do that!

File: README.md (hunk 3 above)
Comment (raquelalegre) [reply README.md]: Whoops. Yes :)

**Comment by raquelalegre** [comment]
@JamieJQuinn have you tried running this locally? I think I'll wait to merge until you do so. We can do that together in the meeting we have scheduled, if you prefer :)

File: README.md (hunk 1 above)
Comment (raquelalegre) [reply README.md]: Oh, actually I just found out that an appropriate version of `npm` gets installed in the `actions/setup-node@v2` workflow step. So probably best to remove the parenthesis.

**Comment by raquelalegre** [comment]
@JamieJQuinn and I had a pair programming sessions and he's gone through the local install well. @JamieJQuinn, are you happy with my latest changes after your review?

**Review by JamieJQuinn — COMMENTED** [review]
Perfect!

**Review by JamieJQuinn — APPROVED** [review]

**Pull request merged**